### PR TITLE
(feat)Composite profile

### PR DESCRIPTION
**Pull request by ranjib**

### CODE REVIEW

File: controller/pwm_profile/diurnal_test.go [hunk 1]
```diff
@@ -6,25 +6,24 @@ import (
 )
 
 func TestDiurnal(t *testing.T) {
-	tFormat := "15:04"
 	conf := `
 {
-	"start":"10:00",
-	"end": "19:30"
+	"start":"10:00:00",
```
Comment (zekth) [inline controller/pwm_profile/diurnal_test.go]: Do format `HH:mm` is deprecated so?

**Comment by ranjib** [comment]
Yes. Planning to delete the auto profile as well. It is now replaced by
“interval” profile. I’ll add details of individual profile type

File: controller/pwm_profile/composite_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "profiles":[
+	  {"type": "diurnal", "span": 300}, {"type": "fixed", "span": 150, "config":{"value": 0}}
+	]
+}
+`
+	p, err := Composite([]byte(conf), time.Now())
+	if err != nil {
+		t.Error(err)
+	}
+	p.Get(time.Now())
```
Comment (Ranthalion) [inline controller/pwm_profile/composite_test.go]: Are there any assertions to make for this test?  It seems to just be testing that composite can unmarshal?

File: controller/pwm_profile/random.go [hunk 3]
```diff
@@ -6,33 +6,44 @@ import (
...
 	t, err := Temporal(conf, min, max)
 	if err != nil {
 		return nil, err
 	}
-	peaks := make([]float64, t.TotalSeconds()/peakInterval)
+	return NewRandom(t), nil
+}
+
+func NewRandom(t temporal) *random {
+	rand.Seed(154)
```
Comment (Ranthalion) [inline controller/pwm_profile/random.go]: Is 154 significant in some way?

Comment (ranjib) [reply controller/pwm_profile/random.go]: Nope. Just a fixed seed number.

Comment (ranjib) [reply controller/pwm_profile/random.go]: will move it as a constant

File: controller/pwm_profile/composite_test.go (hunk 2 above)
Comment (ranjib) [reply controller/pwm_profile/composite_test.go]: yeah.. was done in a hurry. Adding assertions now

File: controller/pwm_profile/composite.go [hunk 4]
```diff
@@ -2,31 +2,84 @@ package pwm_profile
...
+		start = end
 	}
-	return &composite{
-		profiles: profiles,
-	}, nil
+	return &comp, nil
 }
 
 func (c *composite) Get(t time.Time) float64 {
+	reminder := math.Mod(t.Sub(c.start).Seconds(), float64(c.total))
```
[the remark this answers is not in the document]
Comment (ranjib) [reply controller/pwm_profile/composite.go]: Thank you :-) . i'll update

**Pull request merged**